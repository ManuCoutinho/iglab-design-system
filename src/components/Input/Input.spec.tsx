import { screen, render } from '@testing-library/react'
import { Input } from '.'

describe('<Input/>', () => {
  it('should render Input correctly ', () => {
    render(<Input.Input placeholder='test' />)
    expect(screen.getByRole('textbox')).toBeInTheDocument()
    expect(screen.getByRole('textbox')).toHaveAttribute('placeholder', 'test')
  })
  it('should render Input Root correctly ', () => {
    render(<Input.Root>Test</Input.Root>)
    expect(screen.getByText(/test/i)).toBeInTheDocument()
  })
  it('should render Input Icon correctly ', () => {
    render(
      <Input.Icon>
        <p>Icon</p>
      </Input.Icon>
    )
    expect(screen.getByText(/icon/i)).toBeInTheDocument()
  })
  it('match to snapshot', () => {
    const { container } = render(
      <Input.Root>
        <Input.Icon>Icon</Input.Icon>
        <Input.Input />
      </Input.Root>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
