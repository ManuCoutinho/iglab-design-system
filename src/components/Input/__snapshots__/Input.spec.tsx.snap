// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> match to snapshot 1`] = `
<div
  class="h-12 flex items-center gap-3 w-full py-4 px-3 rounded bg-gray-800 focus-within:ring-2 focus-within:outline-cyan-300 focus-within:ring-cyan-500"
>
  <input
    class="outline-0 bg-transparent flex-1 text-gray-100 text-xs placeholder:text-gray-400 "
  />
</div>
`;
